### PR TITLE
chore: MySQL UI generalization

### PR DESCRIPTION
**Pull request by cduflo**
#### Summary

⚠️ **If you're contributing to a plugin please read this section of the [contribution guidelines](https://github.com/cloudquery/cloudquery/blob/main/CONTRIBUTING.md#open-core-vs-open-source) 🧑‍🎓 before submitting this PR** ⚠️

### CODE REVIEW

File: plugins/destination/mysql/cloud-config-ui/src/form/index.tsx [hunk 1]
```diff
@@ -1,188 +1,57 @@
...
-    await handleSubmit(getCurrentValues());
-  };
-
-  const onSubmit = handleFormSubmit(handleTestConnection);
-
-  const parsedTestConnectionError = useMemo(
-    () => (testConnectionError ? parseTestConnectionError(testConnectionError) : undefined),
-    [testConnectionError],
-  );
+  const getCurrentValues = useCallback(() => prepareSubmitValues(form.getValues()), [form]);
```
Comment (jackcloudquery) [inline plugins/destination/mysql/cloud-config-ui/src/form/index.tsx:40]:
```suggestion
  const getCurrentValues = useCallback(() => prepareSubmitValues(getValues()), [getValues]);
```

File: plugins/destination/mysql/cloud-config-ui/src/utils/formSchema.ts [hunk 2]
```diff
@@ -1,88 +1,135 @@
...
+          .default(connectionObj.host ?? ''),
+        port: yup
+          .string()
+          .max(5)
+          .matches(/^($)|(\d+)$/, 'Port must be a number')
+          .default(connectionObj.port ?? ''),
+        database: yup
+          .string()
+          .max(63)
+          .default(connectionObj.database ?? ''),
```
Comment (jackcloudquery) [inline plugins/destination/mysql/cloud-config-ui/src/utils/formSchema.ts:57]: None of those should be required?

**Review by jackcloudquery — APPROVED** [review]
Awesome!

File: plugins/destination/mysql/cloud-config-ui/src/utils/formSchema.ts (hunk 2 above)
Comment (cduflo) [reply plugins/destination/mysql/cloud-config-ui/src/utils/formSchema.ts:57]: Per the go-sql-driver docs, it appears that a DNS string of `/` is valid, making all segments optional.
It is a little confusing in that it states both:
`Except for the databasename, all values are optional. So the minimal DSN is: /dbname`
and 
`If you do not want to preselect a database, leave dbname empty: /`
So the only value that is documented to be both required and optional is dbname 😂 

My take is that all can be optional and the test connection will be the source of truth if the connection gets made. Your thoughts?

ref: https://github.com/go-sql-driver/mysql?tab=readme-ov-file#dsn-data-source-name

Comment (cduflo) [reply plugins/destination/mysql/cloud-config-ui/src/utils/formSchema.ts:57]: @jackcloudquery  - sorry, forgot to tag you

Comment (jackcloudquery) [reply plugins/destination/mysql/cloud-config-ui/src/utils/formSchema.ts:57]: Agreed

**Review by jackcloudquery — APPROVED** [review]

**Pull request merged**